### DISCUSSION TITLE
What is the recommended way to fix scaling in a DE-less i3wm setup?

### DISCUSSION THREAD

**Discussion by pvonmoradi** [i3/i3#6157] [Q&A]
I'm using Debian 12 (bookworm). I've installed i3 meta-package and I'm mostly using Xfce apps (so Gtk3, some Gtk2 and some QT apps).
My screen resolution is 1920×1080. By default UI elements of applications and fonts are too small.
What is the standard way to fix scaling of UI?

- Some Wiki guides recommend [setting `~/.Xresources`](https://wiki.archlinux.org/title/HiDPI#X_Resources). But it mentions that it's better for `Xft.dpi` to be a multiple of 96. For a FHD screen, setting it to 192 makes UI too big. Setting it to lower values means fractional scaling.
- In the same ArchWiki guide, there are environment variables for this: `QT_AUTO_SCREEN_SCALE_FACTOR`, `QT_ENABLE_HIGHDPI_SCALING` and `GDK_SCALE`. This is yet another way to control scaling but possibly conflicts with the above method.

As this problem of mine is not exactly a "HiDPI" issue and FHD screens are very common, I wonder how the community solves this.

**Comment by okraits** [comment]
I'm configuring the GTK font size in the corresponding config files:

https://wiki.archlinux.org/title/GTK#Basic_theme_configuration

I'm using `Sans 10` with a resolution of 2560x1440.

**Reply by pvonmoradi** [reply]
But this doesn't scale UI elements and icons.
